starwars/src/App.js

```diff
@@ -1,41 +1,91 @@
-import React, { Component } from 'react';
-import './App.css';
+import React from 'react';
+import CharacterProfile from './components/CharacterProfile';
 
-class App extends Component {
+class App extends React.Component {
   constructor() {
     super();
     this.state = {
-      starwarsChars: []
+      starwarsChars: [],
+      nextPage: '',
+      previousPage: '',
+      currentPage: `https://swapi.co/api/people/?page=1`
     };
   }
 
+
   componentDidMount() {
-    this.getCharacters('https://swapi.co/api/people');
+    this.getCharacters(this.state.currentPage);
   }
 
   getCharacters = URL => {
-    // feel free to research what this code is doing.
-    // At a high level we are calling an API to fetch some starwars data from the open web.
-    // We then take that data and resolve it our state.
+    // At a high level we are calling an API to fetch some starwars data.
+    // We then take that data and resolve it in our state.
     fetch(URL)
-      .then(res => {
-        return res.json();
-      })
-      .then(data => {
-        this.setState({ starwarsChars: data.results });
-      })
-      .catch(err => {
-        throw new Error(err);
-      });
+    .then(res => {
+      return res.json();
+    })
+    .then(data => {
+      this.setState({ starwarsChars: data.results });
```

Comment:
```this.setState({ starwarsChars: data.results });
      this.setState({ nextPage: data.next });  // Store next page of characters url to nextPage object value.
      this.setState({ previousPage: data.previous });  // Store next page of characters url to nextPage object value.```

So you can actually do all of these setState calls in one line.

```suggestion
      this.setState({ starwarsChars: data.results, nextPage: data.next, previousPage: data.previous });
```
Even though it's not a big deal, it's slightly more performant to do this a single setState call.